extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -159,29 +160,35 @@ export class AzureAccountProvider implements azdata.AccountProvider, vscode.Disp
 					Logger.info(`Could not fetch access token from cache: ${e}, fetching new access token instead.`);
 				}
 				tenantId = tenantId || account.properties.owningTenant.id;
-				let authResult = await azureAuth.getTokenMsal(account.key.accountId, resource, tenantId);
-				if (this.isAuthenticationResult(authResult) && authResult.account && authResult.account.idTokenClaims) {
-					const token: Token = {
-						key: authResult.account.homeAccountId,
-						token: authResult.accessToken,
-						tokenType: authResult.tokenType,
-						expiresOn: authResult.account.idTokenClaims.exp!,
-						tenantId: tenantId,
-						resource: resource
-					};
-					try {
-						await this.msalCacheProvider.writeTokenToLocalCache(token);
-					} catch (e) {
-						Logger.error(`Could not save access token to local cache: ${e}, this might cause throttling of AAD requests.`);
-					}
-					return token;
+				if (getTenantIgnoreList().includes(tenantId)) {
+					// Tenant found in ignore list, don't fetch access token.
+					Logger.info(`Tenant ${tenantId} found in the ignore list, authentication will not be attempted.`);
+					throw new Error(localize('tenantIgnoredError', `Token not acquired as tenant found in ignore list in setting: ${Constants.AzureTenantConfigFilterSetting}`));
```

Comment:
In what situations is this shown to the user? Because there's stuff we have where we iterate over all the tenants in an account - for example it seems every time we request a security token for an account.

https://github.com/microsoft/azuredatastudio/blob/f072a2485c1da6c4daa367bb61f6a725e47e2738/extensions/azurecore/src/account-provider/azureAccountProvider.ts#L229

In which case we shouldn't be throwing, but rather just silently skipping the ignored ones. We should only throw/display an error to the user if an action is explicitly taken to fetch a token for a tenant that is ignored (and even then I'd say we should consider not using the ignore list for those cases)

Reply:
It's only shown when you fetch list of subscriptions or you try to connect to a server and select a tenant that belongs to ignore list.

If we skip showing warning when fetching subscriptions, that does not help user. As there are many cases where users unknowingly select 'Ignore tenant' and the next time they expand account - there would be no subscriptions and no error so user wouldn't know what's going on.

Reply:
How often do we fetch a list of subscriptions though?

If a user purposely ignores a tenant and then it keeps popping up with a toast telling them they ignored it that seems pretty annoying. We should at least add a way for them to suppress future warnings if they really don't care.

Or even better IMO, would be to add an additional prompt when they choose to ignore a tenant asking them to confirm and explaining what that means there. Then we can assume that if they ignore a tenant they know what to expect and won't be surprised by stuff not showing up in the future.

Reply:
> How often do we fetch a list of subscriptions though?

When in 'Browse' tab in connection dialog, and when expanding Azure account in Azure tree. 

> Or even better IMO, would be to add an additional prompt when they choose to ignore a tenant asking them to confirm and explaining what that means there. 

That sounds doable, then we can fully ignore using this tenant for any activity - even populating in tenant list.

Reply:
We also have the getSubscriptions API in azurecore, which other extensions can (and do) use. So yeah, if we can make it so that we don't throw an error in those scenarios that'd be the best I think.